python_modules/dagster/dagster/core/definitions/run_status_sensor_definition.py

```diff
@@ -154,6 +164,73 @@ def failure_event(self):
         return self.dagster_event
 
 
+def build_run_status_sensor_context(
+    sensor_name: str,
+    pipeline_run_status: PipelineRunStatus,
+    dagster_instance: Optional[DagsterInstance] = None,
+    error_info: Optional[SerializableErrorInfo] = None,
+) -> RunStatusSensorContext:
+    """Builds sensor execution context from provided parameters.
+
+    This function can be used to provide the context argument when directly invoking a function
+    decorated with `@run_status_sensor` or `@run_failure_sensor`, such as when writing unit tests.
+
+    If `PipelineRunStatus.FAILURE` is passed for `pipeline_run_status` a `RunFailureSensorContext`
+    will be returned, otherwise a `RunStatusSensorContext` will be returned. These contexts can be
+    converted to a `PipelineFailureContext` using the `to_pipeline_failure()` method
+
+    Args:
+        sensor_name (str): The name of the sensor the context is being constructed for.
+        pipeline_run_status (PipelineRunStatus): The PipelineRunStatus that will trigger the sensor.
+        dagster_instance (Optional[DagsterInstance]): The dagster instance configured for the
+            context. Defaults to DagsterInstance.ephemeral().
+        error_info (Optional[SerializableErrorInfo]): Optional error info to be included in the
+            context.
+
+    Examples:
+    .. code-block:: python
+
+        context = build_run_status_sensor_context(
+            sensor_name="my_sensor",
+            pipeline_run_status=PipelineRunStatus.FAILURE
+        )
+        sensor_to_invoke(context)
+
+    """
+    dagster_event_type = PIPELINE_RUN_STATUS_TO_EVENT_TYPE[pipeline_run_status]
+
+    if dagster_instance is None:
+        dagster_instance = DagsterInstance.ephemeral()
+
+    ctx_cls = RunStatusSensorContext
+    event_specific_data = None
+
+    if dagster_event_type == DagsterEventType.RUN_FAILURE:
+        ctx_cls = RunFailureSensorContext
+        event_specific_data = PipelineFailureData(
+            check.opt_inst_param(error_info, "error_info", SerializableErrorInfo)
+        )
+    elif dagster_event_type == DagsterEventType.RUN_CANCELED:
+        event_specific_data = PipelineCanceledData(
+            check.opt_inst_param(error_info, "error_info", SerializableErrorInfo)
+        )
+
+    dagster_event = DagsterEvent(
+        event_type_value=dagster_event_type.value,
+        pipeline_name="testing_mock",
```

Comment:
This could be provided by the user, or set to some other known mock value. thoughts?